interop.py

```diff
@@ -201,7 +201,7 @@ def run(self):
     
     for server in self._servers:
       for client in self._clients:
-        logging.info("Running with server %s (%s) and client %s (%s)", server, self._servers[server], client, self._clients[client])
+        logging.info("Running with server %s (%s) and client %s (%s)", server, self._implementations[server], client, self._implementations[client])
```

Comment:
You could check the IMPLEMENTATIONS_ROLES dictionary here and skip any implementations that don't have the right roles.

Reply:
When i change code in an existing code base, i always try to stay close to the original design. That's what i did in this case also. I only changed the "clients" and "servers" parameters from dict to list, because the value in their map was redundant w.r.t. the implementations dict (which means that the caller has the responsibility to ensure they are consistent, which is not DRY and may lead to subtle bugs) and the implementation definition was used inconsistently in this class: in the creating the command to run, the implementation was taken from the "implementation" parameter, only in this log statement the implementation was taken form the "servers" and "clients" parameters.

I think your proposal is not so much commenting my change, but a change or improvement in itself and as such imho this should not be part of this PR.